lib/dash/mpd_utils.js

```diff
@@ -268,6 +268,10 @@ shaka.dash.MpdUtils = class {
     const durationStr =
         MpdUtils.inheritAttribute(context, callback, 'duration');
     let segmentDuration = XmlUtils.parsePositiveInt(durationStr || '');
+    const ContentType = shaka.util.ManifestParserUtils.ContentType;
+    if (context.representation.contentType == ContentType.IMAGE) {
+      segmentDuration = XmlUtils.parseFloat(durationStr || '');
```

Comment:
The problem here is that many content uses durations with decimals. If you use parsePositiveInt it doesn't return anything, if I use parseFloat it returns this correct, but it doesn't pass all the tests. That's why I only use it for images. @joeyparrish , do you have a more elegant solution to this?

Reply:
I believe the DASH spec says the segment duration attribute needs to be an integer.  If you want a partial second, use `timescale`.

Reply:
I added the comment.